content/blog/2021/06/15/LetsOptaPlanYourJbpmTasksPart2.adoc

```diff
@@ -0,0 +1,131 @@
+
+. Your application can simply use these components when needed, typically by using CDI and OptaPlanner standard APIs. All the rest is managed by OptaPlanner runtime.
+
+In the following image we show a simplified diagram of the *task-assigning-service* project, which contains the *taskAssigningDefaultSolverConfig.xml* (Solver configuration), the *TaskAssigningSolution* (PlanningSolution), and the *TaskAssignment* and *User* (PlanningEntities) and includes the *optaplanner-quarkus* extension.
+
+See how these elements can be used in your code:
+
+image::TaskAssigningServiceApplication.png[]
+
+=== The Task Assigning Service
+
+Finally, we can see all the building blocks working together:
+
+
+image::TaskAssigningServiceArchitecture.png[]
+
+When the CreditDispute process is executed the following actions takes place:
+
+. A process instance is created by using the corresponding endpoints.
+
+. When the human tasks are created/completed/etc. Cloud Events are produced and delivered to Kafka thanks to the inclusion of the kogito-events-reactive-messaging-addon.
+
+. Different https://kogito.kie.org/[Kogito] supporting services like the Data Index Service or the Task Assigning Service receives these events.
+
+. At the Task Assigning Service, OptaPlanner takes immediate control on these events, and the new tasks assignment plan is calculated and optimized.
+
+. Finally, by using the generated process endpoints the necessary assignments are produced.
+
+This cycle is repeated over and over.
+
+== Process execution
+
+Let's now execute the process in the https://kogito.kie.org/[Kogito] ecosystem.
+
+image::CreditDisputeProcessDiagramWithTaskInputs.png[]
+
+The following image shows how a new process instance is created by using the Kogito generated endpoints:
+
+image::Start_CreditDispute_1111_MASTER_Spanish.png[]
+
+Analogous the link:../../../2020/07/03/LetsOptaPlanYourJbpmTasksPart1.html[Let's OptaPlan your jBPM tasks (part 1)] case, this information combined with the user's configuration is used to establish which of the available users are best suited for each task.
+
+This time, the following user's configuration is used:
+
+[cols="10%,40%,50%" frame="all", options="header"]
+|===
+|User
+|Belongs the group
+|Has skills in
+
+|katy | CreditAnalyst | CITI and MASTER cards.
+|emily | CreditAnalyst | VISA and AMERICAN_EXPRESS cards.
+|bob | ClientRelations | Spanish and English languages.
+|frank | ClientRelations | Chinese language.
+
+|===
+
+When the process is started a new instance of the ResolveDispute task is created and automatically assigned:
+
+image::ResovleDispute_1111_MASTER_Spanish_katy.png[]
+
+We can see that the task requires "a CreditAnalyst user with skills in the MASTER credit card". This is why it was assigned to katy.
```

Comment:
katy with a capital?

Writing docs guidelines is use active tense, which works well in blogs too:
"we can see that the task requires" => "The task requires ..."